Analysis/src/TypeFunction.cpp

```diff
@@ -2206,6 +2206,19 @@ TypeFunctionReductionResult<TypeId> indexFunctionImpl(
         return {std::nullopt, true, {}, {}};
 
     TypeId indexerTy = follow(typeParams.at(1));
+
+    // Whether the type is still pending.
+    // 
+    // TODO: A generalized sanity check function for all type functions?
+    // But it should be optional for customization
+    // for functions that specifically need type function
+    // This here fixes a stack overflow.
+    // https://github.com/luau-lang/luau/issues/1406
```

Comment:
Can you please remove all of this comment? `isPending` _is_ the "generalized sanity check" that should be called for each type parameter to a type function.

Reply:
Alright, give me a second